### PR TITLE
session: split `TestInitializeSQLFile` into four ones to speed up

### LINKED ISSUE #46832: slow test `TestInitializeSQLFile` in `session` pkg
## Bug Report

<details><summary><h3>Average timecost was 298850.0000ms between "2023-09-01" AND "2023-09-08"<h3></summary>

Found records: 1  (filter only for runs that timecost >= threshold)
Max timecost: 298850ms
Min timecost: 298850ms
Avg timecost: 298850.0000ms

- [CI build](https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/ghpr_unit_test/32112/): used 298850ms

</details>

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #46832

Problem Summary:

Four test cases for 'bootstrap initialize SQL file' were integrated into one case, so its speed is slow.

### What is changed and how it works?

1. Split the test case into four ones
2. Use `intest.InTest` for limiting `failToLoadOrParseSQLFile`

### Check List

Tests 

- [x] Unit test (previous tests)
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by CbcWestwolf** [comment]
/cc lcwangchao 
PTAL

**Review by fzzf678 — APPROVED** [review]

File: session/bootstrap_test.go [hunk 1]
```diff
@@ -947,29 +947,22 @@ func TestUpgradeToVer85(t *testing.T) {
 	MustExec(t, se, "delete from mysql.bind_info where default_db = 'test'")
 }
 
-func TestInitializeSQLFile(t *testing.T) {
```
Comment (hawkingrei) [inline session/bootstrap_test.go:950]: Can you remove this test case into ```session/test```?

**Review by hawkingrei — APPROVED** [review]

File: session/bootstrap.go [hunk 2]
```diff
@@ -1000,6 +1000,13 @@ var internalSQLTimeout = owner.ManagerSessionTTL + 15
 // whether to run the sql file in bootstrap.
 var runBootstrapSQLFile = false
 
+// DisableRunBootstrapSQLFileInTest only used for test
+func DisableRunBootstrapSQLFileInTest() {
```
Comment (Defined2014) [inline session/bootstrap.go:1004]: Maybe move it to another file is better. ( no need reviewer by `sig-critical-approvers-tidb-server`

**Pull request merged**